factory/bootstrap/bootstrapComponentsHandler.go

```diff
@@ -117,12 +119,25 @@ func (mbf *managedBootstrapComponents) RoundActivationHandler() process.RoundAct
 	return mbf.bootstrapComponents.roundActivationHandler
 }
 
+// SetShardCoordinator sets the shard coordinator
+func (mbf *managedBootstrapComponents) SetShardCoordinator(shardCoordinator sharding.Coordinator) error {
+	mbf.mutBootstrapComponents.RLock()
+	defer mbf.mutBootstrapComponents.RUnlock()
+
+	if mbf.bootstrapComponents == nil {
+		return errors.ErrNilBootstrapComponents
+	}
+
+	mbf.bootstrapComponents.shardCoordinator = shardCoordinator
```

Comment:
should the shardCoordinator be allowed nil? checkIfNil and add a unit test maybe?

Reply:
moved this method on a test component